src/lints/enum_variant_deprecated.ron

```diff
@@ -0,0 +1,60 @@
+SemverQuery(
+    id: "enum_variant_deprecated",
+    human_readable_name: "enum variant marked with #[deprecated]", 
+    description: "An enum variant was marked with the #[deprecated] attribute.",
+    required_update: Minor,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        enum_name: name @output
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                        variant {
+                            variant_name: name @output @tag
+                            deprecated @filter(op: "=", value: ["$true"])  
+                            span_: span @optional {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
+                    }
+                }
+            }
+            baseline {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                        variant {
+                            name @filter(op: "=", value: ["%variant_name"])
+                            deprecated @filter(op: "!=", value: ["$true"])  
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+    },
+    error_message: "An enum variant was marked as deprecated. Downstream crates will get a compiler warning when using this variant.",
+    per_result_error_template: Some("variant {{enum_name}}::{{variant_name}} in {{span_filename}}:{{span_begin_line}}"),
+    witness: (
+        hint_template: r#"match value {
+    {{enum_name}}::{{variant_name}} => { /* ... */ },
+    _ => { /* ... */ }
+}"#
+    ),
+)
```

Comment:
Please configure your text editor to always add a trailing newline to all files, and ensure all files in the PR have trailing newlines.